server/src/main/java/io/crate/execution/ddl/tables/DropConstraintRequest.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Licensed to Crate.io GmbH ("Crate") under one or more contributor
+ * license agreements.  See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.  Crate licenses
+ * this file to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.  You may
+ * obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ * However, if you have executed another commercial license agreement
+ * with Crate these terms will supersede the license and you may use the
+ * software solely pursuant to the terms of the relevant commercial agreement.
+ */
+
+package io.crate.execution.ddl.tables;
+
+import io.crate.metadata.RelationName;
+import org.elasticsearch.action.support.master.AcknowledgedRequest;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+
+public class DropConstraintRequest extends AcknowledgedRequest<DropConstraintRequest> {
+
+    private final RelationName relationName;
+    private final boolean isPartitioned;
```

Comment:
Is it necessary to provide this information? I think it would be possible to derive this based on the RelationName and clusterState

Reply:
I tried to remove the flag and compute it as
```
Index[] concreteIndices = resolveIndices(currentState, request.relationName().indexNameOrAlias());
boolean isPartitioned = false;
for (Index index: concreteIndices) {
    if (IndexParts.isPartitioned(index.getName())) {
        isPartitioned = true;
        break;
    }
}
```

but test started to fail as `concreteIndices` was empty - I guess it might work only on non-empty tables but otherwise we need this flag?

Reply:
You could check if there's a template in the cluster state that matches the relationName.

The reason I'm asking is also because of https://github.com/crate/crate/issues/11939

Reply:
done in https://github.com/crate/crate/pull/12996/commits/11e662e481a48dca69d5810262083f7637f2b1e2, thanks